src/app/thankYou/thankYou.component.js

```diff
@@ -43,6 +49,14 @@ class ThankYouController{
         if(this.purchase.paymentMeans.self.type === 'elasticpath.purchases.purchase.paymentmeans'){ //only credit card type has billing address
           this.billingAddress = this.orderService.formatAddressForTemplate(this.purchase.paymentMeans['billing-address'].address);
         }
+
+        // Display Account Benefits Modal when GUEST checkout
```

Comment:
Yes, it immediately shows the modal when the user is a PUBLIC or has been matched. I was told the Account Benefits inside the thank-you page are if the user accidentally dismisses the modal. They both need to be present.